### PR TITLE
:rewind::fast_forward: Allow result limits to be set by client

### PR DESCRIPTION
**Pull request by st3v3nhunt**
In this commit I've moved most of the checking of input into the middleware rather than doing it in both middleware and lib. I've ummed and ahhed about whether I should take out the validation from the lib but I'm not a fan of having it duplicated and don't believe the lib is the right place for it so I have moved it...
There is still work to be done on this code - i'm thinking specifically to stop passing `geo` from middleware to lib. And I think the tests could be condensed down further.

### CODE REVIEW

File: app/middleware/getPharmacies.js [hunk 1]
```diff
@@ -7,31 +7,34 @@ function getPharmacies(req, res, next) {
...
     log.warn(errors, 'getPharmacies errors');
     res.status(400).json(errors);
+  } else {
+    const latitude = req.query.latitude;
+    const longitude = req.query.longitude;
+    const nearby = req.query['limits:results:nearby'] || 3;
+    const open = req.query['limits:results:open'] || 1;
+    // Given how search performance is impacted by the radius of the search
+    // it has intentionaly not been allowed to be specificed by the client
+    const searchRadius = 20;
```
Comment (neilbmclaughlin) [inline app/middleware/getPharmacies.js]: Might be worth relaxing this in future so the searchRadius is passed as a parameter but is validated to be less than some system configurable value.

File: test/integration/app.js [hunk 2]
```diff
@@ -62,6 +61,35 @@ describe('app', () => {
...
+          expect(nearby).to.be.instanceof(Array);
+          expect(open).to.be.instanceof(Array);
+          expect(nearby.length).to.be.equal(nearbyResults);
+          expect(open.length).to.be.equal(openResults);
+          done();
+        });
+    });
   });
 
   describe('nearby error handling', () => {
```
Comment (neilbmclaughlin) [inline test/integration/app.js]: This is more of a general observation.

The structure of the tests that I write differs subtly from yours.

In the example here:
```
describe 'nearby error handling'
  it 'when the open limit is supplied and is not a number',
    expect ...
    expect ...
```
 
I would try to write as something like
```
describe 'nearby error handling'
  describe 'when the open limit is supplied and is not a number',
    it 'should return a json error message'
      expect ...
      expect ...
    it 'should specify the expected range'
      expect  ...
```

Although functionally the same the difference comes when considering the tests as documentation of the code. The output for the test run for the first example doesn't describe the code behaviour. I'd say that 'it' should be followed by 'should' to describe behaviour.

[Update: just spotted that the getPharmacies tests or structured using describe/it/should]

@st3v3nhunt @mcheung-nhs - any thoughts on this?

**Comment by st3v3nhunt** [comment]
Damn difference in package version strikes again :rage1:

**Comment by neilbmclaughlin** [comment]
@teneightfive & @marcofucci  had problems with shrinkwrap - I think it was an Azure issue.

Do you know what they were and are we likely to hit them?

**Comment by st3v3nhunt** [comment]
@neilbmclaughlin Is there anything beyond the failing build due to the lint issue that needs fixes on this?

**Comment by st3v3nhunt** [comment]
Possibly something when we deploy to azure. Lets find out...

**Pull request merged**